roboflow/util/model_processor.py

```diff
@@ -0,0 +1,312 @@
+            )
+
+        print_warn_for_wrong_dependencies_versions([("ultralytics", ">=", "8.3.0")], ask_to_continue=True)
+
+    model = torch.load(os.path.join(model_path, filename))
+
+    if isinstance(model["model"].names, list):
+        class_names = model["model"].names
+    else:
+        class_names = []
+        for i, val in enumerate(model["model"].names):
+            class_names.append((val, model["model"].names[val]))
+        class_names.sort(key=lambda x: x[0])
+        class_names = [x[1] for x in class_names]
+
+    if "yolov8" in model_type or "yolov10" in model_type or "yolov11" in model_type:
+        # try except for backwards compatibility with older versions of ultralytics
+        if "-cls" in model_type or model_type.startswith("yolov10") or model_type.startswith("yolov11"):
+            nc = model["model"].yaml["nc"]
+            args = model["train_args"]
+        else:
+            nc = model["model"].nc
+            args = model["model"].args
+        try:
+            model_artifacts = {
+                "names": class_names,
+                "yaml": model["model"].yaml,
+                "nc": nc,
+                "args": {k: val for k, val in args.items() if ((k == "model") or (k == "imgsz") or (k == "batch"))},
+                "ultralytics_version": ultralytics.__version__,
+                "model_type": model_type,
+            }
+        except Exception:
+            model_artifacts = {
+                "names": class_names,
+                "yaml": model["model"].yaml,
+                "nc": nc,
+                "args": {
+                    k: val for k, val in args.__dict__.items() if ((k == "model") or (k == "imgsz") or (k == "batch"))
+                },
+                "ultralytics_version": ultralytics.__version__,
+                "model_type": model_type,
+            }
+    elif "yolov5" in model_type or "yolov7" in model_type or "yolov9" in model_type:
+        # parse from yaml for yolov5
+
+        with open(os.path.join(model_path, "opt.yaml")) as stream:
+            opts = yaml.safe_load(stream)
+
+        model_artifacts = {
+            "names": class_names,
+            "nc": model["model"].nc,
+            "args": {
+                "imgsz": opts["imgsz"] if "imgsz" in opts else opts["img_size"],
+                "batch": opts["batch_size"],
+            },
+            "model_type": model_type,
+        }
+        if hasattr(model["model"], "yaml"):
+            model_artifacts["yaml"] = model["model"].yaml
+
+    with open(os.path.join(model_path, "model_artifacts.json"), "w") as fp:
+        json.dump(model_artifacts, fp)
+
+    torch.save(model["model"].state_dict(), os.path.join(model_path, "state_dict.pt"))
+
+    list_files = [
+        "results.csv",
+        "results.png",
+        "model_artifacts.json",
+        "state_dict.pt",
+    ]
+
+    zip_file_name = "roboflow_deploy.zip"
+
+    with zipfile.ZipFile(os.path.join(model_path, zip_file_name), "w") as zipMe:
```

Comment:
Not for now, but we should stop zipping these big files with Python. It is too slow for big files.